exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/exporter.py

```diff
@@ -282,81 +271,71 @@ def _translate_attributes(self, attributes) -> TypingSequence[KeyValue]:
     def _export(
         self, data: Union[TypingSequence[ReadableSpan], MetricsData]
     ) -> ExportResultT:
+        try:
+            self._export_backoff(data)
+        except RpcError:
+            return self._result.FAILURE
+
+    @staticmethod
+    def _on_backoff(details):
+        this = details["args"][0]
+        _, error, _ = sys.exc_info()
+        assert isinstance(error, RpcError)
+        wait_delay = details["wait"]
+        logger.warning(
+            ("Transient error %s encountered while exporting " "%s, retrying in %ss."),
+            error.code(),
+            this._exporting,
+            round(wait_delay, 1),
+        )
 
-        # FIXME remove this check if the export type for traces
-        # gets updated to a class that represents the proto
-        # TracesData and use the code below instead.
-        # logger.warning(
-        #     "Transient error %s encountered while exporting %s, retrying in %ss.",
-        #     error.code(),
-        #     data.__class__.__name__,
-        #     delay,
-        # )
-        max_value = 64
-        # expo returns a generator that yields delay values which grow
-        # exponentially. Once delay is greater than max_value, the yielded
-        # value will remain constant.
-        for delay in expo(max_value=max_value):
-
-            if delay == max_value:
-                return self._result.FAILURE
+    @backoff.on_exception(backoff.expo, RpcError, max_time=60, on_backoff=_on_backoff)
```

Comment:
Does it have to be a decorator? If I wanted to configure the max_time from the exporter instance how would one do that?